### PR TITLE
[TACHYON-1453] Automatically distribute Tachyon to Yarn nodes

### PR DESCRIPTION
**Pull request by aaudiber**
https://tachyon.atlassian.net/browse/TACHYON-1453

### CODE REVIEW

**Comment by aaudiber** [comment]
Jenkins, test this please.

**Comment by aaudiber** [comment]
Jenkins, test this please.

File: integration/bin/tachyon-yarn-setup.sh [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+if [ $CONTAINER_TYPE = 'tachyon-master' ]; then
+  ./integration/bin/tachyon-master-yarn.sh $@
+elif [ $CONTAINER_TYPE = 'tachyon-worker' ]; then
+  ./integration/bin/tachyon-worker-yarn.sh $@
+elif [ $CONTAINER_TYPE = 'application-master' ]; then
+  ./integration/bin/tachyon-application-master.sh $@
+else
+  echo "Unrecognized container type: $CONTAINER_TYPE"
+  exit 1
+fi
```
Comment (yupeng9) [inline integration/bin/tachyon-yarn-setup.sh]: newline?

File: integration/bin/tachyon-yarn-submit.sh [hunk 2]
```diff
@@ -0,0 +1,60 @@
+#!/bin/bash
+#
+# Usage:
+#  tachyon-yarn-submit.sh <numWorkers> <pathHdfs>
+
+function printUsage {
+  echo "Usage: tachyon-yarn-submit.sh <numWorkers> <pathHdfs>"
+  echo -e "  numWorkers        \tNumber of Tachyon workers to launch"
+  echo -e "  pathHdfs          \tpath on HDFS to put tachyon jar and distribute it to YARN"
```
Comment (yupeng9) [inline integration/bin/tachyon-yarn-submit.sh]: Capitalize `path` to be consistent with the option above

File: integration/bin/tachyon-yarn-submit.sh [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+echo "Starting YARN client to launch Tachyon on YARN"
+
+# Add Tachyon java options to the yarn options so that tachyon.yarn.Client can be configured via
+# tachyon java options
+export YARN_OPTS="${YARN_OPTS:-${TACHYON_JAVA_OPTS}}"
+
+${HADOOP_HOME}/bin/yarn jar ${JAR_LOCAL} tachyon.yarn.Client \
+    -num_workers $NUM_WORKERS \
+    -master_address localhost \
+    -resource_path ${HDFS_PATH}
```
Comment (yupeng9) [inline integration/bin/tachyon-yarn-submit.sh]: newline

File: integration/yarn/src/test/java/tachyon/yarn/ApplicationMasterTest.java [hunk 4]
```diff
@@ -196,7 +224,7 @@ public Void answer(InvocationOnMock invocation) throws Throwable {
    * already used by other workers. This tests {@link ApplicationMaster} as a whole, only mocking
    * its clients.
    */
-  @Test(timeout = 10000)
+  @Test(timeout = 100000)
```
Comment (yupeng9) [inline integration/yarn/src/test/java/tachyon/yarn/ApplicationMasterTest.java]: Do we need 100s for timeout?

**Comment by yupeng9** [comment]
Thanks for this big improvement. I left some minor comments.

File: integration/yarn/src/test/java/tachyon/yarn/ApplicationMasterTest.java (hunk 4 above)
Comment (aaudiber) [inline integration/yarn/src/test/java/tachyon/yarn/ApplicationMasterTest.java]: Good catch, I forgot to change it back after debugging

File: integration/bin/tachyon-yarn-setup.sh (hunk 1 above)
Comment (aaudiber) [inline integration/bin/tachyon-yarn-setup.sh]: fixed, thanks

File: integration/bin/tachyon-yarn-submit.sh (hunk 2 above)
Comment (aaudiber) [inline integration/bin/tachyon-yarn-submit.sh]: done, thanks

File: integration/bin/tachyon-yarn-submit.sh (hunk 3 above)
Comment (aaudiber) [inline integration/bin/tachyon-yarn-submit.sh]: done, thanks

**Comment by apc999** [comment]
Should we rename `integration/bin/tachyon-yarn-submit.sh` to `integration/bin/tachyon-yarn.sh`? The reason I created another file is because back then I didn't want to modify an existing working script. Since now I guess the new launcher script is better than the old one, we could probably simply overwrite the old  script. The advantage is people used the old script don't need to search what happened to a missing file after updating Tachyon. What do you think?

File: integration/bin/tachyon-yarn-setup.sh [hunk 5]
```diff
@@ -0,0 +1,19 @@
+#!/bin/bash
+
+# prepare Tachyon dir structure for a YARN container to launch master or worker
```
Comment (apc999) [inline integration/bin/tachyon-yarn-setup.sh]: Maybe add a comment saying that, this script is referenced as 
`tachyon.yarn.Utils#TACHYON_SETUP_SCRIPT` in source code.
So in the future if we rename this script, we should also update the source code.

**Comment by yupeng9** [comment]
LGTM

**Comment by apc999** [comment]
+1 LGTM. I left a minor comment, if you prefer the current approach, I will merge the PR then.

**Comment by apc999** [comment]
+1 LGTM

**Pull request merged**